sale_block_cancellation/models/sale.py

```diff
@@ -0,0 +1,111 @@
+# coding: utf-8
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2016 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: Luis Torres <luis_t@vauxoo.com>
+#    planned by: Sabrina Romero <sabrina@vauxoo.com>
+############################################################################
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    @api.depends('picking_ids', 'picking_ids.state', 'state')
+    def _get_allow_cancel(self):
+        '''If all the pickings related to a SO are on "Ready to transfer" or
+        in "Waiting another move" state, the cancellation of this must be
+        allowed, or even whit pickings on different states, if the moved
+        product quantities are correctly returned.'''
+        for sale in self:
+            sale.allow_cancel = False
```

Comment:
By default the boolean fields are False, you can exclude this to avoid call the write method twice for each order